cloudwatch_metric_alarm: fix idempotency for alarm without dimensions

jmisset-cb opened the pull request:
##### SUMMARY
A metric alarm in Cloudwatch can optionally have Dimensions. When a metric alarm in Cloudwatch does not have any dimensions, it returns: "Dimensions": [] when queried via boto3.

When configuring a metric alarm without Dimensions in Cloudwatch using the cloudwatch_metric_alarm plugin, Dimensions must be absent from the parameters.

Because "Dimensions": [] does not match Dimensions: None, the result is always Changed. 

This Pull Request fixes this by setting Dimensions from the returned alarm parameters to None when the field is empty.

Fixes #1750

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
cloudwatch_metric_alarm

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

jmisset-cb commented:
@gravesm Is there anything I can do to get someone to review this PR + the related issue it solves? Thanks in advance.

GomathiselviS commented on tests/integration/targets/cloudwatch_metric_alarm/tasks/main.yml:
```diff
@@ -492,6 +492,44 @@
+        evaluation_periods: 3
+        unit: Percent
+        description: This will alarm when an instance's cpu usage average is lower than
+          5% for 15 minutes
```
Can you add an assertion task here?

GomathiselviS commented:
Hi @jmisset-cb Thank you for initiating this pull request. We are currently in the process of reviewing it. Kindly incorporate a changelog entry and perform a rebase of the PR.

jmisset-cb commented:
Hi @GomathiselviS thank you for taking the time to review my PR.
I've added the assertion task and merge the latest main commits into the branch.
Could you please assist me with the changelog? In which file/directory should I place this?
Thanks in advance!

GomathiselviS commented:
> Hi @GomathiselviS thank you for taking the time to review my PR. I've added the assertion task and merge the latest main commits into the branch. Could you please assist me with the changelog? In which file/directory should I place this? Thanks in advance!

[This](https://github.com/ansible-collections/amazon.aws/blob/main/changelogs/fragments/1883-cloudwatchevent_rule-fix-json-input-handling-for-input_template.yml) an example of a changelog fragment. You can create one for your fix. The category of your fix will be one of [these](https://github.com/ansible-collections/amazon.aws/blob/main/changelogs/config.yaml#L12)

jmisset-cb commented:
> > Hi @GomathiselviS thank you for taking the time to review my PR. I've added the assertion task and merge the latest main commits into the branch. Could you please assist me with the changelog? In which file/directory should I place this? Thanks in advance!
> 
> [This](https://github.com/ansible-collections/amazon.aws/blob/main/changelogs/fragments/1883-cloudwatchevent_rule-fix-json-input-handling-for-input_template.yml) an example of a changelog fragment. You can create one for your fix. The category of your fix will be one of [these](https://github.com/ansible-collections/amazon.aws/blob/main/changelogs/config.yaml#L12)

I've added the changelog entry! Thanks again.

GomathiselviS commented on plugins/modules/cloudwatch_metric_alarm.py:
```diff
@@ -334,6 +334,10 @@ def create_metric_alarm(connection, module, params):
+
         # Exclude certain props from change detection
         for key in [
             "ActionsEnabled",
```
Can `Dimensions` be added to this list, instead of handling it separately?

tremble replied:
yes

alinabuzachis reviewed (commented):
@jmisset-cb Thank you for taking time to work on this bugfix and for patience. Can you please rebase your Pull Request?

jmisset-cb commented:
@alinabuzachis No worries, thank you for looking into this. I have rebased the PR.
